test/attr/attr_completionhandlerasync.swift

```diff
@@ -0,0 +1,129 @@
+// REQUIRES: concurrency
+
+// RUN: %target-typecheck-verify-swift -enable-experimental-concurrency
+
+// ===================
+// Parsing
+// ===================
+
+func asyncFunc() async -> Int { }
+
+@completionHandlerAsync("asyncFunc()", completionHandlerIndex: 1)
+func goodFunc1(value: String, completionHandler: @escaping (Int) -> Void) {}
+
+@completionHandlerAsync("asyncFunc()")
+func goodFunc2(value: String, completionHandler: @escaping (Int) -> Void) {}
+
+// expected-error@+1:24{{expected '(' in 'completionHandlerAsync' attribute}}
+@completionHandlerAsync
+func func1() {}
+
+// expected-error@+1:25{{argument of 'completionHandlerAsync' attribute must be an identifier or full function name}}
+@completionHandlerAsync("not+identifier")
+func func2() {}
+
+// expected-error@+1:25{{argument of 'completionHandlerAsync' attribute must be an identifier or full function name}}
+@completionHandlerAsync("$dollarname")
+func func3() {}
+
+// expected-error@+1:25{{argument of 'completionHandlerAsync' attribute must be an identifier or full function name}}
+@completionHandlerAsync("TypePrefix.func")
+func func4() {}
+
+// expected-error@+1{{argument of 'completionHandlerAsync' cannot be an interpolated string literal}}
+@completionHandlerAsync("interpreted \()")
+func func5() {}
+
+@completionHandlerAsync("foo" // expected-error{{expected ')' in 'completionHandlerAsync' attribute}}
+func func6() {}
+
+@completionHandlerAsync("foo", completionHandlerIndex: 2 // expected-error@:57{{expected ')' in 'completionHandlerAsync'}}
+func func7() {}
+
+// expected-error@+1{{'@completionHandlerAsync' attribute cannot be applied to this declaration}}
+@completionHandlerAsync("foo", completionHandlerIndex: 0)
+protocol SomeProto {
+  // expected-error@+1:27{{no suitable '@completionHandlerAsync' async functions 'protoFunc'}}
+  @completionHandlerAsync("protoFunc", completionHandlerIndex: 0)
+  func protoFunc(continuation: @escaping () -> Void)
+}
+
+// expected-error@+1{{'@completionHandlerAsync' attribute cannot be applied to this declaration}}
+@completionHandlerAsync("foo", completionHandlerIndex: 0)
+struct SomeStruct: SomeProto {
+  // expected-error@+1:27{{no suitable '@completionHandlerAsync' async functions 'protoFunc'}}
```

Comment:
The message sort of makes it sound like there should be another `protoFunc` that is `async` and attributed with `@completionHandlerAsync`. What do you think about `no corresponding async function named 'protoFunc' for 'protoFunc'`? Maybe "async version of function" or "async alternative function" even.